server/authz/authz.go

```diff
@@ -119,6 +119,30 @@ func (a *Authorizer) Authorize(ctx context.Context, object, action interface{})
 	return nil
 }
 
+func (a *Authorizer) TeamAuthorize(ctx context.Context, teamID uint, action string) error {
```

Comment:
Up until now, we weren't really checking that a user could make changes or read a specific team, so I added this to use now alongside the regular `Authorize`